### PR TITLE
fix(device): Escape the ModelId in the MQTT layer

### PR DESCRIPTION
**Pull request by drwill-ms**
The report from Stefan based on some testing is that when using AMQP this value is escaped, but not for .NET. Other langs (C) have reported always escaping this, and requested that we do too.

The model Id currently can't have characters that would require escaping, so for now this really isn't an issue. However, for self-describing devices there is talk of using a query-string-like approach which would add escapable characters. For MQTT the model Id already sits in the path in the query string so it can't have an embedded query string too.

### CODE REVIEW

**Review by jamdavi — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

**Review by azabbasi — APPROVED** [review]

File: iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs [hunk 1]
```diff
@@ -344,7 +344,7 @@ private async Task ConnectAsync(IChannelHandlerContext context)
                 // the SDK will enable plug and play features by appending the model Id to the MQTT CONNECT packet (in the username).
                 if (!(_options?.ModelId).IsNullOrWhiteSpace())
                 {
-                    usernameString += $"&{ModelIdParam}={_options.ModelId}";
+                    usernameString += $"&{ModelIdParam}={Uri.EscapeDataString(_options.ModelId)}";
```
Comment (rido-min) [inline iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs:347]: should we escape just the querysting?
I would not expect `dtmi:rido:model;1?hash=C32C4` to be encoded as `dtmi%3Arido%3Amodel%3B1%3Fhash%3DC32C4` rather `dtmi:rido:model;1%3Fhash%3DC32C4`

**Review by bikamani — APPROVED** [review]

File: iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs (hunk 1 above)
Comment (bikamani) [reply iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs:347]: What's the value look like for Amqp?

Comment (drwill-ms) [reply iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs:347]: As long as it is unencoded on the service side, it shouldn't matter how it is encoded over the wire.

Comment (drwill-ms) [reply iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs:347]: But I'm taking the question seriously and making sure we're doing the right thing. My understanding is that the whole value is being escaped in the C SDK.

Comment (drwill-ms) [reply iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs:347]: Confirmed whole value is escaped in the C SDK, and arrives at the Hub twin intact (aka unescaped). I'll confirm for this change and then merge.

Comment (drwill-ms) [reply iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs:347]: Yep, works fine.

**Pull request merged**